File: CoreList/App/TG/View/TGCell.m
//
//  TGCell.m
//  CoreList
//
//  Created by 成林 on 15/6/7.
//  Copyright (c) 2015年 muxi. All rights reserved.
//

#import "TGCell.h"
#import "TGModel.h"

@interface TGCell ()

@property (weak, nonatomic) IBOutlet UILabel *idLabel;

@property (weak, nonatomic) IBOutlet UILabel *titleLabel;

@property (weak, nonatomic) IBOutlet UILabel *descLabel;

@property (weak, nonatomic) IBOutlet UILabel *contentLabel;

@property (weak, nonatomic) IBOutlet UILabel *typeLabel;


@end



@implementation TGCell

- (void)awakeFromNib {
    // Initialization code
    self.layer.borderColor = [UIColor lightGrayColor].CGColor;
    self.layer.borderWidth = 2.5f;
}




-(void)dataFill:(TGModel *)tgModel{
    
    _idLabel.text = [NSString stringWithFormat:@"%@",@(tgModel.hostID)];
    
    _titleLabel.text = tgModel.title;
    
    _descLabel.text = tgModel.about;
    
    _contentLabel.text = tgModel.content;
    
    _typeLabel.text = [NSString stringWithFormat:@"%@",@(tgModel.type)];
}


@end

File: CoreList/App/TG/View/TGCell.swift
//
//  TGCell.swift
//  CoreList
//
//  Created by 冯成林 on 16/1/4.
//  Copyright © 2016年 muxi. All rights reserved.
//

import UIKit


class TGCell: CoreListCollectionViewCell {

    @IBOutlet weak var idLabel: UILabel!
    
    @IBOutlet weak var tgTitleLabel: UILabel!
    
    @IBOutlet weak var descLabel: UILabel!
    
    @IBOutlet weak var contentLabel: UILabel!
    
    @IBOutlet weak var typeLabel: UILabel!
    
    static var rid: String = "TGCell"
}

extension TGCell {

    override func awakeFromNib() {
        
        super.awakeFromNib()
        
        layer.borderColor = UIColor.gray.cgColor
        layer.borderWidth = 0.5
    }
    
    override class func dequeueReusableCell(with collectionView: UICollectionView!, indexPath: IndexPath!) -> TGCell! {
        
        return collectionView.dequeueReusableCell(withReuseIdentifier: rid, for: indexPath) as! TGCell
    }
    
    override func dataFill(_ coreModel: CoreModel!) {
        
        let tgModel = coreModel as! TGModel
        
        idLabel.text = "\(tgModel.hostID)"
        
        tgTitleLabel.text = tgModel.title;
        
        contentLabel.text = tgModel.content;
        
        typeLabel.text = "\(tgModel.score)"
    }
}
